src/models/user.js

```diff
@@ -73,13 +73,13 @@ export default {
     removeFavoriteController(state, { payload }) {
       return {
         ...state,
-        favoriteControllers: state.favoriteControllers.filter(item => item.key !== payload.key),
+        favoriteControllers: state.favoriteControllers.filter(item => item !== payload),
```

Comment:
@anishaswain rather than doing referential equality here can we use some function that does "deep equality" like lodash's isDeepEqual(). I am being a little paranoid for edge cases here and I may be wrong